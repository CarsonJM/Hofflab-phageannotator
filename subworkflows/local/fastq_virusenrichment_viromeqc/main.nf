//
// Estimate viral enrichment in reads
//
include { VIROMEQC_INSTALL  } from '../../../modules/local/viromeqc/install/main'
include { VIROMEQC_VIROMEQC } from '../../../modules/local/viromeqc/viromeqc/main'

workflow FASTQ_VIRUSENRICHMENT_VIROMEQC {
    take:
    fastq_gz    // [ [ meta.id ] , [ reads_1.fastq.gz, reads_2.fastq.gz ]   , reads (mandatory)
    viromeqc_db // [ viromeqc_db ]                                          , viromeqc database (optional)

    main:
    ch_versions = Channel.empty()

    // if viromeqc_db exists, skip VIROMEQC_INSTALL
    if ( viromeqc_db ){
        ch_viromeqc_db  = viromeqc_db
    } else {
        //
        // MODULE: download viromeQC database
        //
        ch_viromeqc_db  = VIROMEQC_INSTALL( ).viromeqc_index
        ch_versions     = ch_versions.mix(VIROMEQC_INSTALL.out.versions)
    }

    //
    // MODULE: Estimate viral enrichment
    //
    ch_enrichment_tsv   = VIROMEQC_VIROMEQC(fastq_gz, ch_viromeqc_db.first()).enrichment
    ch_versions         = ch_versions.mix(VIROMEQC_VIROMEQC.out.versions)

    emit:
    enrichment_tsv  = ch_enrichment_tsv // [ [ meta.id ] , enrichment.tsv ]  , TSV containing viral enrichment estimates
    versions        = ch_versions       // [ versions.yml ]
}
