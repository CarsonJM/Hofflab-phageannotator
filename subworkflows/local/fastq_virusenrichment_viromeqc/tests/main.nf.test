nextflow_workflow {

    name "Test subworkflow: FASTQ_VIRUSENRICHMENT_VIROMEQC"
    script "../main.nf"
    workflow "FASTQ_VIRUSENRICHMENT_VIROMEQC"

    setup {
        run("VIROMEQC_INSTALL") {
            script "../../../../modules/local/viromeqc/install/main.nf"
        }
    }

    test("fastq.gz + no_db") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                        ]
                    ],
                    [
                        [ id:'test' ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                        ]
                    ]
                )
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("fastq.gz + db") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                        ]
                    ],
                    [
                        [ id:'test' ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                        ]
                    ]
                )
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
