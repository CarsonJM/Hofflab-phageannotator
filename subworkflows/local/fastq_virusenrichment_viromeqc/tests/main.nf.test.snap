{
    "fastq.gz + db": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "1": [
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a",
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a"
                ],
                "enrichment_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a",
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T10:02:45.357637842"
    },
    "fastq.gz + no_db": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4ccc4c0f15f5c15071a00dd99999869d",
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a",
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a"
                ],
                "enrichment_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4ccc4c0f15f5c15071a00dd99999869d",
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a",
                    "versions.yml:md5,68b364ec64f345aa4158d184a84a896a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T09:57:45.344847295"
    }
}